model/src/main/kotlin/config/DownloaderConfiguration.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+import org.ossreviewtoolkit.model.SourceCodeOrigin
+import org.ossreviewtoolkit.spdx.getDuplicates
+
+data class DownloaderConfiguration(
+    /**
+     * Configuration of the considered source code origins and their priority order.
+     */
+    val sourceCodeOrigins: List<SourceCodeOrigin> = listOf(SourceCodeOrigin.VCS, SourceCodeOrigin.ARTIFACT)
+) {
+    init {
+        require(sourceCodeOrigins.isNotEmpty()) {
+            "'sourceCodeOrigins' must not be empty."
+        }
+
+        val duplicates = sourceCodeOrigins.getDuplicates()
+        require(duplicates.isEmpty()) {
+            "'sourceCodeOrigins' must not contain duplicates. Duplicates: $duplicates"
+        }
```

Comment:
Shouldn't we drop this check in favor of re-introducing `distinct()` in the later commit?

Reply:
See: https://github.com/oss-review-toolkit/ort/pull/3736#discussion_r601156237